### ISSUE TITLE
Show primary error message in error header

### ISSUE THREAD

**Issue by justinmchase** [jenkinsci/github-checks-plugin#200]
### Feature Request

When I report an error like this in one of my steps:
```groovy
stage ("Submodule Check") {
  steps {
    script {
      error "Submodules are OK for branches but cannot be merged into main"
    }
  }
}
```

This plugin does not report that error cleanly in the check. I expect it to show the message I reported more prominently but instead it shows information about the error step itself.

For example right in the Pull Request instead of showing that error message it simply says `error in 'error' step`:

![Screen Shot 2021-09-04 at 8 10 00 AM](https://user-images.githubusercontent.com/10974/132095706-42ca196d-40c8-4434-a174-d23121b88228.png)

Other github status checkers will show the actual error message in this part of the UI instead.

Additionaly when going to the error details it has the same message prominently in a header as well as some redundant information about an Error in the error step.

(red for bad, green for good)
![Screen Shot 2021-09-04 at 8 11 19 AM](https://user-images.githubusercontent.com/10974/132095868-aca41b76-8290-478d-a280-c9f4b1c89e3c.png)

I feel like in the case of reporting an error via the `error()` step / function, the information about that step should be omitted and the error message itself should be what is reported.

**Comment by justinmchase** [comment]
Also, the `unstable` step has the same issue:

![Screen Shot 2021-09-04 at 10 25 49 AM](https://user-images.githubusercontent.com/10974/132099692-f5be6aea-fa71-49af-9c05-5b5b58d9809a.png)

Ideally it would show the actual message when in the unstable state as well.

**Comment by KalleOlaviNiemitalo** [comment]
The information about failing steps is formatted in [FlowExecutionAnalyzer.java](https://github.com/jenkinsci/checks-api-plugin/blob/83a4321ed0d4ae69cfa8e2bb34dbfe7b428556dd/src/main/java/io/jenkins/plugins/checks/status/FlowExecutionAnalyzer.java), so this issue belongs to [jenkinsci/checks-api-plugin](https://github.com/jenkinsci/checks-api-plugin).
